compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1298,11 +1298,16 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
    TR::Register *falseReg = cg->evaluate(falseVal);
    bool trueValIs64Bit = TR::TreeEvaluator::getNodeIs64Bit(trueVal, cg);
    TR::Register *trueReg  = TR::TreeEvaluator::intOrLongClobberEvaluate(trueVal, trueValIs64Bit, cg);
+   if (!node->isNotCollected()) 
+      trueReg->setContainsCollectedReference();
 
    // don't need to test if we're already using a compare eq or compare ne
    auto conditionOp = condition->getOpCode();
+   bool longCompareOn32bit = (conditionOp.isBooleanCompare() &&
```

Comment:
when is the condition op not a Boolean compare or not treated as a Boolean compare - this seems redundant...

Reply:
It seems in other places where the code deals with ternary node, it doesn't assume the first child is a boolean compare - e.g. Evaluator on Power OMR::Power::TreeEvaluator::iternaryEvaluator, and OMR::CodeGenerator::rematerializeCmpUnderTernary in lower trees. If the first child is an instnaceOf we would need the isBooleanCompare check.

Reply:
correct - the first child can be any node that generates a suitable Boolean value to drive the test